### PR TITLE
Use Custom serializer

### PR DESCRIPTION
**Pull request by santi6291**
The built-in `ArraySerializer` adds the data attributes to includes in transformers. This custom serializer solves this issue by only setting the `resourceKey` to the included data if set rather than falling back to `data`

https://github.com/thephpleague/fractal/issues/430#issuecomment-461130026

### CODE REVIEW

File: src/serializer/AlgoliaSerializer.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     *
+     * @param string $resourceKey
+     * @param array  $data
+     *
+     * @return array
+     */
+    public function collection($resourceKey, array $data)
+    {
+        if ($resourceKey === false) {
+            return [$resourceKey => $data];
```
Comment (riasvdv) [inline src/serializer/AlgoliaSerializer.php]: Shouldn't this be reversed? If the resourceKey is false then just return the data?

**Comment by riasvdv** [comment]
Can you give an example on how you're using this in Scout? And if you're comfortable with it add a test to check if the output of the serializer is exactly what we'd expect?

**Comment by santi6291** [comment]
Hi, the use case is when we have a transformer that uses Fractal include.
```
class ProductTransformer extends TransformerAbstract
{
    /**
     * List of resources possible to include
     *
     * @var array
     */
    protected $defaultIncludes = [
        'productSpecs',
    ];
    
    ...

    /**
     * @return \League\Fractal\Resource\Item
     */
    public function includeProductSpecs()
    {
        return $this->collection($this->getAttr('productSpecs'), new ProductSpecsTransformer);
    }
}

class ProductSpecsTransformer extends TransformerAbstract 
{
    public function transform(\craft\commerce\elements\Product $product)
    {
        return [
            'title' => (string) $this->getAttr('title'),
        ];
    }   
}
```

the following code would send the following to Algolia
```
{
    "title": "blah blah"
    "productSpecs": {
        "data": [{
            "title": "foo"
        }, {
            "title": "bar"
        }]
    }

}
```

With the change on this PR the output to Algolia would remove the extra `data` attribute from the included field

```
{
    "title": "blah blah"
    "productSpecs": [{
        "title": "foo"
    }, {
        "title": "bar"
    }]
}
```

I'll take a stab at writing something to check the output of the Fractal toArray doesn't return a `resourceKey` unless is defined

**Comment by riasvdv** [comment]
I've never actually used Fractal beyond a basic array, so thank you for explaining the use-case. Any code I merge will become mine to maintain so I make sure to understand why it's there.

The test would be great, as it would've caught the error I commented on ;)

**Comment by santi6291** [comment]
Hi, totally agree! I started taking a stab at writing the test but I can't seem to get codecept to execute the test, any tip as to what i could be missing?

**Comment by riasvdv** [comment]
@santi6291 There should be a `/** @test **/`comment right before your test functions

**Comment by santi6291** [comment]
Hi @riasvdv added the test, let me know if there is anything else I can do here?

**Comment by riasvdv** [comment]
Awesome, looks good! Thank you for your work

**Pull request merged**